LISISO/install.sh

```diff
@@ -87,6 +81,94 @@ if [[ $distro_version != "5"* ]]; then
+        fi
+else
+        echo "This is re-install/upgrade"
+        #Same Partition Calculation
+        same_partition_minimum_req_disk=$(bc <<< $rpm_install_size+$min_initramfs+$temp_disk_usage_same_partition_reinstall)
+        #Different Partition Calculation
+        diff_partition_minimum_req_disk_boot=$(expr $min_initramfs + $temp_disk_usage_diff_partition_boot_reinstall)
+        diff_partition_minimum_req_disk_module=$(bc <<< $rpm_install_size+$temp_disk_usage_diff_partition_modules_reinstall)
+        if [ "$distro_version" == 73 ] || [ "$distro_version" == 74 ]; then
+                same_partition_minimum_req_disk=$(bc <<< $same_partition_minimum_req_disk+$mlx_driver_size)
+                diff_partition_minimum_req_disk_module=$(bc <<< $diff_partition_minimum_req_disk_module+$mlx_driver_size)
+        fi
+fi
+
+#
+# Main script body
+#
+# Check if kernel version running is higher compared to installed kernel.
+# This is important because kmod are dependent on Kernel version
+
+disk_space_errors=0
+boot_partition_name=$(df -hm  /boot | awk '{print $1}' | tail -1)
+modules_partition_name=$(df -hm /lib/modules | awk '{print $1}' | tail -1)
+if [[ $boot_partition_name == $modules_partition_name ]]; then
+        echo  "boot and modules share the same partition."
+        current_boot_partition_size=`df -hm $boot_partition_name | awk '{print $4}' | tail -1`
+        echo "$boot_partition_name Required size: $same_partition_minimum_req_disk MB. Available: $current_boot_partition_size"
+        if [[ ( $current_boot_partition_size -lt $same_partition_minimum_req_disk ) ]]; then
+                echo "ERROR: Please make sure that you have sufficient disk space available in '$boot_partition_name'."
+                disk_space_errors=$(expr $disk_space_errors + 1)
+        fi
+else
+        echo  "boot and modules do not share the same partition."
+        current_boot_partition_space=`df -hm | grep "/boot" | awk '{print $4}'`
+        current_modules_partition_space=`df -hm /lib/modules | awk '{print $4}' | tail -1`
+	echo "$boot_partition_name Required size: $diff_partition_minimum_req_disk_boot MB. Available: $current_boot_partition_space"
+        echo "$modules_partition_name Required size: $diff_partition_minimum_req_disk_module MB. Available: $current_modules_partition_space"
+        if [ ${current_boot_partition_space%.*} -lt ${diff_partition_minimum_req_disk_boot%.*} ]; then
+                echo "ERROR: Please make sure that you have sufficient disk space available in '$boot_partition_name'."
+                disk_space_errors=$(expr $disk_space_errors + 1)
+        fi
+        if [ ${current_modules_partition_space%.*}  -lt ${diff_partition_minimum_req_disk_module%.*} ]; then
+                echo "ERROR: Please make sure that you have sufficient disk space available in '$modules_partition_name'."
+                disk_space_errors=$(expr $disk_space_errors + 1)
+        fi
+fi
```

Comment:
This should be implemented as a function and it should be just called before install